packages/e2e-tests/test/e2e-current-op.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { expect } from 'chai';
+import {
+  skipIfApiStrict,
+  startSharedTestServer,
+} from '../../../testing/integration-testing-hooks';
+import { TestShell } from './test-shell';
+
+describe('e2e currentOp', function () {
+  skipIfApiStrict();
+  afterEach(TestShell.cleanup);
+
+  const testServer = startSharedTestServer();
+
+  context('with 2 shells', function () {
+    let helperShell: TestShell;
+    let currentOpShell: TestShell;
+
+    const OPERATION_TIME = 1_000;
+    const CURRENT_OP_WAIT_TIME = 100;
+    this.timeout(OPERATION_TIME * 2);
+
+    beforeEach(async function () {
+      helperShell = TestShell.start({
+        args: [await testServer.connectionString()],
+      });
+      currentOpShell = TestShell.start({
+        args: [await testServer.connectionString()],
+      });
+      await helperShell.waitForPrompt();
+
+      // Insert a dummy object so find commands will actually run with the delay.
+      await helperShell.executeLine('db.coll.insertOne({})');
+    });
+
+    it('should return the correct operation', async function () {
+      const regexOperation = helperShell.executeLine(
+        `db.coll.find({$where: function() { sleep(${OPERATION_TIME}) }}).projection({test: 1})`
+      );
+      helperShell.assertNoErrors();
+      await currentOpShell.executeLine(`sleep(${CURRENT_OP_WAIT_TIME})`);
```

Comment:
I could use an actual JS timeout-based sleep here.

Reply:
Yeah, that might just be a bit easier 🙂